cirq/contrib/qasm_import/_parser.py

```diff
@@ -267,14 +287,13 @@ def p_gate_op_with_params(self, p):
 
     def _resolve_gate_operation(self, args: List[List[ops.Qid]], gate: str,
                                 p: Any, params: List[float]):
-        if gate not in self.basic_gates.keys():
-            raise QasmException('Unknown gate "{}" at line {}, '
-                                'maybe you forgot to include '
-                                'the standard qelib1.inc?'.format(
-                                    gate, p.lineno(1)))
-        p[0] = self.basic_gates[gate].on(args=args,
-                                         params=params,
-                                         lineno=p.lineno(1))
+        helper_text = (", did you forget to include qelib1.inc?"
```

Comment:
Don't compute error text on success branches. Move this inside the failure case.

(Why isn't it inlined into the message in the first place?)

Reply:
Hmm, yeah, this is pretty gnarly, I rushed it on my commute this morning. I'll move it inline to the error message.